pkg/build/api/v1/conversion.go

```diff
@@ -6,9 +6,45 @@ import (
 
 	oapi "github.com/openshift/origin/pkg/api"
 	newer "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+func convert_v1_BuildConfig_To_api_BuildConfig(in *BuildConfig, out *newer.BuildConfig, s conversion.Scope) error {
+	if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
+		return err
+	}
+
+	newTriggers := []newer.BuildTriggerPolicy{}
+	// strip off any default imagechange triggers where the buildconfig's
+	// "from" is not an ImageStreamTag, because those triggers
+	// will never be invoked.
+	imageRef := buildutil.GetImageStreamForStrategy(out.Spec.Strategy)
+	hasIST := imageRef != nil && imageRef.Kind == "ImageStreamTag"
+	for _, trigger := range out.Spec.Triggers {
+		if trigger.Type != newer.ImageChangeBuildTriggerType {
+			newTriggers = append(newTriggers, trigger)
+			continue
+		}
+		if trigger.ImageChange.From == nil {
+			if !hasIST {
```

Comment:
yeah it should be, i think this was a victim of refactoring along the way.  i'll change it.